server/controllers/items/recent_public.coffee

```diff
@@ -22,20 +22,28 @@ module.exports = (req, res)->
   sanitize req, res, sanitization
   .then (params)->
     { assertImage, lang, limit } = params
-    items_.publicByLangAndDate itemsQueryLimit, offset, lang, assertImage, reqUserId
-    .then selectRecentItems(limit)
+    items_.publicByDate itemsQueryLimit, offset, assertImage, reqUserId
+    .then selectRecentItems(lang, limit)
   .then bundleOwnersToItems.bind(null, res, reqUserId)
   .catch error_.Handler(req, res)
 
-selectRecentItems = (limit)-> (items)->
-  _(items)
-  .groupBy itemsOwnerId
-  .map (ownerItems, _)-> ownerItems.slice(0, maxItemsPerOwner)
-  .flatten()
-  .values()
-  .slice 0, limit
-  .shuffle()
-  # wrapping lodash .chain() function
-  .value()
+selectRecentItems = (lang, limit)-> (items)->
+  recentItems = []
+  discardedItems = []
+  itemsCountByOwner = {}
+
+  for item in items
+    if recentItems.length is limit then return recentItems
+    if item.snapshot['entity:lang'] is lang and itemsCountByOwner[item.owner] < 3
```

Comment:
if `itemsCountByOwner[item.owner]` is `undefined`, `itemsCountByOwner[item.owner] < 3` is `false`, so initialization (`itemsCountByOwner[item.owner] ?= 0`) should be done before this line